src/main/blackbox/blackbox.c

```diff
@@ -914,6 +914,39 @@ static bool sendFieldDefinition(const char * const *headerNames, unsigned int he
 }
 
 /**
+ * Encode and transmit PID
+ */
+static uint32_t blackboxWritePID(uint8_t pid)
+{
+    uint8_t kP, kI, kD;
+
+    if (IS_PID_CONTROLLER_FP_BASED(currentProfile->pidProfile.pidController)) {
+        if (pid < 3) {
+            kP = constrain(lrintf(currentProfile->pidProfile.P_f[pid] * 10.0f), 0, 250);
```

Comment:
There is no need to constrain the values at all. It creates additional processing overhead.
And even if you would use constrain you need to use constrainf for floats instead of constrain